src/machine/lib_machine.rs

```diff
@@ -235,6 +235,10 @@ impl Machine {
             called: false,
         }
     }
+
+    pub fn cleanup_query_iter(&mut self) {
+        self.trust_me()
```

Comment:
Doing this explicitly shouldn't be necessary as it happens automatically on drop https://github.com/mthom/scryer-prolog/pull/2465/files#diff-52044e834bac7030da1f0c418d81dc4773d33df32ab26dd0b6f6913175900c2eR31

Reply:
Yes, and because of the rules of Rust, you have to drop QueryState before being able to use Machine again, so this is indeed guaranteed to always run before you use the machine after a query.

Reply:
Not necessarily, drop isn't guaranteed as one could just [forget](https://doc.rust-lang.org/std/mem/fn.forget.html) the QueryState

Reply:
Fair point, and this detail made some interfaces unsound even in the standard library in the past. I thought about that when implementing this actually, but I couldn't figure out an interface with better guarantees that wouldn't be extremely clunky to use. I just assume that nobody (at least in the Rust side) is going to put this somewhere where it leaks. I also don't know how `trust_me` actually works or if it is even the right thing to do to clean the machine state, so I'm not confident this is actually the right guarantee to make. But assuming it is, this should definitely be documented in the item documentation when we get around to that.